### DISCUSSION TITLE
Ability to convert JenkinsFile to GitHub actions offline

### DISCUSSION THREAD

**Discussion by ilyesAj** [github/gh-actions-importer#193] [Feature requests] [answered]
**why ?**
In some environments where the connectivity is limited, we should be able to convert any Jenkinsfile to GitHub actions 

**what ?**
Dry-run command should run without the need to specify `--jenkins-instance-url` or `--source-url` . you should not need to authenticate to GitHub actions.

**expected:**
```
gh actions-importer dry-run jenkins --source-file-path backend/deployment/Jenkinsfile -o output
```
-> the result would be to output the converted file

**Comment by Vichugo1995** [comment]
Ok

**Comment by ethanis** [comment, marked as answer]
@ilyesAj a connection to the Jenkins server is required, even when using the `--source-file-path` option, so that Actions Importer can convert the Jenkinsfile contents to a JSON object that can then be converted to its equivalent workflow using this [endpoint](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/EXTENDING.md#conversion-to-json-representation-from-jenkinsfile).

**Comment by OpenByteDev** [comment]
that requirement seems very odd to me
